Content.Server/Health/BodySystem/Surgery/SurgeryData/BiologicalSurgeryData.cs

```diff
@@ -11,81 +12,125 @@
-                            return DisconnectOrganSurgery;
-                        else if (toolType == SurgeryToolType.Cauterization)
-                            return CautizerizeIncisionSurgery;
-                    }
-                    else
-                    {
-                        if (toolType == SurgeryToolType.Retraction)
-                            return RetractSkinSurgery;
-                        else if (toolType == SurgeryToolType.Cauterization)
-                            return CautizerizeIncisionSurgery;
-                    }
-                }
-                else
-                {
-                    if (toolType == SurgeryToolType.VesselCompression)
-                        return ClampVesselsSurgery;
-                    else if (toolType == SurgeryToolType.Cauterization)
-                        return CautizerizeIncisionSurgery;
-                }
+                return RemoveBodyPartSurgery;
             }
-            else
+            if (!_skinOpened) //Case: skin is normal.
             {
-                if (toolType == SurgeryToolType.Incision)
+                if (toolType == SurgeryType.Incision)
                     return OpenSkinSurgery;
             }
+            else if (_skinOpened && !_vesselsClamped) //Case: skin is opened, but not clamped.
+            {
+                if (toolType == SurgeryType.VesselCompression)
+                    return ClampVesselsSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
+            else if (_skinOpened && _vesselsClamped && !_skinRetracted) //Case: skin is opened and clamped, but not retracted.
+            {
+                if (toolType == SurgeryType.Retraction)
+                    return RetractSkinSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
+            else if (_skinOpened && _vesselsClamped && _skinRetracted) //Case: skin is fully open.
+            {
+                if (_parent.Mechanisms.Count > 0 && toolType == SurgeryType.VesselCompression && (_disconnectedOrgans.Except(_parent.Mechanisms).Count() != 0 || _parent.Mechanisms.Except(_disconnectedOrgans).Count() != 0))
+                    return LoosenOrganSurgery;
+                else if (_disconnectedOrgans.Count > 0 && toolType == SurgeryType.Incision)
+                    return RemoveOrganSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
             return null;
         }
 
-        protected void OpenSkinSurgery(BodyManagerComponent target, IEntity performer)
+        public override string GetDescription()
+        {
+            string toReturn = "";
+            if (_skinOpened && !_vesselsClamped) //Case: skin is opened, but not clamped.
```

Comment:
again, see above about avoiding repeating yourself here.

Reply:
In this case its necessary for both checks since _skinOpened and _vesselsClamped can both be true, in which case we want a different message. With just a _skinOpened check it would return the first message even if _vesselsClamped is also true.

Reply:
Uh, no, it wouldn't.

You factor out the `if(_skinOpened)` and then you do your if on `_vesselsClamped` true/false, and then within `if_vesselsClamped)` you do one final if on `_skinRetracted`

you still get unique messages for all cases.